backend/src/middleware/hashtags/extractHashtags.js

```diff
@@ -1,26 +1,27 @@
 import cheerio from 'cheerio'
+import { exec, build } from 'xregexp/xregexp-all.js'
 // formats of a Hashtag:
 //   https://en.wikipedia.org/w/index.php?title=Hashtag&oldid=905141980#Style
 // here:
 //    0. Search for whole string.
-//    1. Hashtag has only 'a-z', 'A-Z', and '0-9'.
-//    2. If it starts with a digit '0-9' than 'a-z', or 'A-Z' has to follow.
-const ID_REGEX = /^\/search\/hashtag\/(([a-zA-Z]+[a-zA-Z0-9]*)|([0-9]+[a-zA-Z]+[a-zA-Z0-9]*))$/g
+//    1. Hashtag has only all unicode characters and '0-9'.
+//    2. If it starts with a digit '0-9' than a unicode character has to follow.
+const regX = build('^/search/hashtag/((\\pL+[\\pL0-9]*)|([0-9]+\\pL+[\\pL0-9]*))$')
 
 export default function(content) {
   if (!content) return []
   const $ = cheerio.load(content)
   // We can not search for class '.hashtag', because the classes are removed at the 'xss' middleware.
-  //   But we have to know, which Hashtags are removed from the content es well, so we search for the 'a' html-tag.
+  //   But we have to know, which Hashtags are removed from the content as well, so we search for the 'a' html-tag.
   const urls = $('a')
     .map((_, el) => {
       return $(el).attr('href')
     })
     .get()
   const hashtags = []
   urls.forEach(url => {
-    let match
-    while ((match = ID_REGEX.exec(url)) != null) {
```

Comment:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-09-03T13:56:50Z" title="Tuesday, September 3rd 2019, 3:56:50 pm +02:00">Sep 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -20,7 +21,7 @@ export default function(content) {
   const hashtags = []
   urls.forEach(url => {
     let match
-    while ((match = ID_REGEX.exec(url)) != null) {
+    if ((match = exec(url, regX)) != null) {
```


```suggestion
    if (match != null) {
```

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-09-03T13:59:33Z" title="Tuesday, September 3rd 2019, 3:59:33 pm +02:00">Sep 3, 2019</time>_

---


now that this isn't a `while loop` anymore we can actually make `match` a `const` and define it 🙂

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-09-04T12:10:00Z" title="Wednesday, September 4th 2019, 2:10:00 pm +02:00">Sep 4, 2019</time>_

---


Thanks you @alina-beck !!!
Nice to hear.

Yes, an improvement! 🤗😍
@mattwr18 did already … 😍